core/src/main/java/hudson/model/Job.java

```diff
@@ -1351,7 +1351,7 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             final ProjectNamingStrategy namingStrategy = Jenkins.get().getProjectNamingStrategy();
                 if (namingStrategy.isForceExistingJobs()) {
-                    namingStrategy.checkName(name);
+                    namingStrategy.checkName(getParent().getFullName(),name);
```

Comment:
Just to comply with Checkstyle

```suggestion
                    namingStrategy.checkName(getParent().getFullName(), name);
```